src/main/java/com/teamacronymcoders/epos/locks/keys/tag/ParentTagLockKey.java

```diff
@@ -0,0 +1,77 @@
+package com.teamacronymcoders.epos.locks.keys.tag;
+
+import com.teamacronymcoders.epos.api.EposAPI;
+import com.teamacronymcoders.epos.api.locks.keys.IParentLockKey;
+import com.teamacronymcoders.epos.api.requirements.IRequirement;
+import java.util.ArrayList;
+import java.util.Collection;
+import java.util.List;
+import javax.annotation.Nonnull;
+import javax.annotation.Nullable;
+import net.minecraft.block.Block;
+import net.minecraft.block.BlockState;
+import net.minecraft.fluid.Fluid;
+import net.minecraft.fluid.FluidState;
+import net.minecraft.item.Item;
+import net.minecraft.item.ItemStack;
+import net.minecraft.nbt.CompoundNBT;
+import net.minecraft.util.ResourceLocation;
+import net.minecraftforge.fluids.FluidStack;
+
+public class ParentTagLockKey implements IParentLockKey {
+
+    @Nonnull
+    private final Collection<ResourceLocation> tags;
+    @Nullable
+    private final CompoundNBT nbt;
+
+    public ParentTagLockKey(@Nonnull Collection<ResourceLocation> tags, @Nullable CompoundNBT nbt) {
+        this.tags = tags;
+        this.nbt = nbt == null || nbt.isEmpty() ? null : nbt;
+    }
+
+    @Nullable
+    public static ParentTagLockKey fromObject(@Nonnull Object object) {
+        if (object instanceof ItemStack) {
+            ItemStack stack = (ItemStack) object;
+            if (stack.isEmpty()) {
+                return null;
+            }
+            return fromTags(stack.getItem().getTags(), stack.getTag());
+        } else if (object instanceof Item) {
+            return fromTags(((Item) object).getTags());
+        } else if (object instanceof BlockState) {
+            return fromTags(((BlockState) object).getBlock().getTags());
+        } else if (object instanceof Block) {
+            return fromTags(((Block) object).getTags());
+        } else if (object instanceof FluidStack) {
+            FluidStack stack = (FluidStack) object;
+            return fromTags(stack.getFluid().getTags(), stack.getTag());
+        } else if (object instanceof Fluid) {
+            return fromTags(((Fluid) object).getTags());
+        } else if (object instanceof FluidState) {
+            return fromTags(((FluidState) object).getFluid().getTags());
+        }
+        return null;
```

Comment:
Add support for EntityType tags (the final type of built in tag in MC that this doesn't already support)

Reply:
If you're gonna do things regarding tags, you should add a way for an addon to register a Function<T,Tags>

Reply:
By this do you mean that they would be registering a function such as `Function<Item, Tag<Item>>` or what do you have in mind by this.

Reply:
For now I left a TODO to go back at some point and look into letting mods register other types of Tags as the system probably needs some work once we come up with more potential use cases.